sinks/src/main/scala/com.snowplowanalytics.snowplow.eventgen/Main.scala

```diff
@@ -104,20 +109,61 @@ object Main extends IOApp {
       if (predicate) sinkBuilder(name, idx)
       else _.drain
 
+    def makeOutput: Pipe[F, GenOutput, Unit] =
+      if (outputDir.toString.startsWith("kinesis:")) {
+        if (List(config.withRaw, config.withEnrichedTsv, config.withEnrichedJson).count(identity) > 1)
+          throw new RuntimeException(s"Kinesis could only output in single format")
+        if (config.compress) {
+          throw new RuntimeException(s"Kinesis doesn't support compression")
+        }
+        val kinesisClient: KinesisAsyncClient = KinesisClientUtil.createKinesisAsyncClient(KinesisAsyncClient.builder())
+
+        def reqBuilder(event: Event): PutRecordRequest = PutRecordRequest.builder()
+          .streamName(outputDir.getRawAuthority)
+          .partitionKey(event.event_id.toString)
+          .data(SdkBytes.fromUtf8String(event.toTsv))
+          .build()
+
+        st: Stream[F, GenOutput] =>
+          st.map(_._2)
+            .flatMap(Stream.emits)
+            .map(reqBuilder)
+            .parEvalMap(10)(e =>
+              Async[F].fromCompletableFuture(Sync[F].delay(kinesisClient.putRecord(e)))
+            ).void
```

Comment:
It would be great to put batch records inside `PutRecords` [like in Enrich](https://github.com/snowplow/enrich/blob/3.7.0/modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Sink.scala#L272-L284) for instance, for better throughput

Reply:
I agree that it'd be better that way - what we have here does work just fine for the load tests I did. I've made [an issue](https://github.com/snowplow-incubator/snowplow-event-generator/issues/44) for this improvement.